# interactive-has-name

Interactive elements MUST has name.

> For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies.

[WCAG 2.1 - 4.1.2: Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value)

## Options

### `ignore: string | string[]`

**Default:** None

Ignores the element specified in the selector string from the validation target.

## :white_check_mark: Correct

```html
<button type="button">button</button>
```

```html
<span id="label">button</span>
<button type="button" aria-labelledby="label"></button>
```

```html
<input type="text" aria-label="input" />
```

```html
<div id="div" tabindex="0">Element</div>
```

If the element is not exposed to the accessibility API, name allows an empty.

```html
<div aria-hidden="true">
  <button></button>
</div>
```

**Example:** `ignore: '[data-testid="aware-invalid-button"]'`

You can use the `ignore` option to ignore certain buttons.

```html ignore:"[data-testid='aware-invalid-button']"
<button data-testid="aware-invalid-button"></button>
```

## :warning: Incorrect

```html
<button></button>
```

```html
<a href="#"><img src="" alt="" /></a>
```

```html
<button><span style="display: none">button</span></button>
```

```html
<div id="div"></div>

<script>
  const div = document.getElementById('div');
  const handler = () => {
    /* ... */
  };
  div.addEventListener('click', handler, false);
  div.addEventListener('keydown', handler, false);
</script>
```
