# link-has-name

Link MUST has name.

> The purpose of each link can be determined from the link text alone or from the link text together with its programmatically determined link context, except where the purpose of the link would be ambiguous to users in general.

[WCAG 2.1 - 2.4.4 Link Purpose (In Context)](https://www.w3.org/TR/WCAG21/#link-purpose-in-context)

> For all user interface components (including but not limited to: form elements, links and components generated by scripts), the name and role can be programmatically determined; states, properties, and values that can be set by the user can be programmatically set; and notification of changes to these items is available to user agents, including assistive technologies.

[WCAG 2.1 - 4.1.2 Name, Role, Value](https://www.w3.org/TR/WCAG21/#name-role-value)

## :white_check_mark: Correct

```html
<a href="https://www.w3.org/TR/WCAG21/#name-role-value"
  >WCAG 2.1 - 4.1.2 Name, Role, Value</a
>
```

## :warning: Incorrect

```html
<a href="https://www.w3.org/TR/WCAG21/#name-role-value"></a>

<style>
  a::after {
    content: 'WCAG 2.1 - 4.1.2 Name, Role, Value'/'';
  }
</style>
```
